### PR TITLE
Add retry strategy for AU-NT (NETSMO parser)

### PR DESCRIPTION
**Pull request by unitrium**
## Issue

NETSMO parser has been down for a few days. It seems that the first call to the website always end with a 502 error at the moment. Strangely, retrying to get the website works fine afterwards.

## Description
Adds a retry strategy to NETSMO parser so that it can avoid those issues.

### Preview

image
Getting data with test_parser.

### CODE REVIEW

**Review by mathilde-daugy — APPROVED** [review]

**Comment by unitrium** [comment]
@FelixDQ I reworked it with the decorator as discussed. Thanks for your input!

File: parsers/NTESMO.py [hunk 1]
```diff
@@ -60,6 +60,15 @@ class Generator(TypedDict):
...
 }
 
+# For some reason the page doesn't always load on first attempt.
+# Therefore we retry a few times.
+retry_strategy = Retry(
+    total=3,
+    status_forcelist=[500, 502, 503, 504],
+)
+session_with_retries = Session()
+session_with_retries.mount("https://", HTTPAdapter(max_retries=retry_strategy))
```
[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply parsers/NTESMO.py]: The recent change makes this comment way less relevant so resolving this, but I think we should keep it in mind if we want to move forward with a parser class.

File: parsers/lib/config.py [hunk 2]
```diff
@@ -19,3 +23,21 @@ def wrapped_f(*args, **kwargs):
...
+
+def retry_policy(retry_policy: Retry):
+
+    assert isinstance(retry_policy, Retry)
+
+    def wrap(f):
+        def wrapped_f(*args, **kwargs):
+            session = args[1] if len(args) > 2 else kwargs.get("session")
+            session = Session() if session is None else session
+            session.mount("https://", HTTPAdapter(max_retries=retry_policy))
```
Comment (FelixDQ) [inline parsers/lib/config.py:36]: Should we mount it on `http://` as well?

**Review by FelixDQ — COMMENTED** [review]
Great! Thinking about edge cases here:
* What happens if you call the parser with a session that already has a mounted adapter? Would this just override that or result in an error?
* Could we potentially detect if the given session already contains an adapter and log a warning saying that we have overwritten the adapter?

**Pull request merged**